dali/operators/reader/numpy_reader_op.cc

```diff
@@ -223,18 +223,41 @@ void NumpyReaderCPU::RunImpl(HostWorkspace &ws) {
     if (need_transpose_[i])  // 2x if transposition is required
       task_sz *= 2;
 
-    thread_pool.AddWork([&, i](int tid) {
-      if (need_slice_[i] && need_transpose_[i]) {
+    if (need_slice_[i] && need_transpose_[i]) {
+      thread_pool.AddWork([&, i](int tid) {
         SlicePermuteHelper(output[i], file_i.data, rois_[i], fill_value_);
-      } else if (need_slice_[i]) {
+      }, task_sz);
+    } else if (need_slice_[i]) {
+      thread_pool.AddWork([&, i](int tid) {
         SliceHelper(output[i], file_i.data, rois_[i], fill_value_);
-      } else if (need_transpose_[i]) {
+      }, task_sz);
+    } else if (need_transpose_[i]) {
+      thread_pool.AddWork([&, i](int tid) {
         TransposeHelper(output[i], file_i.data);
+      }, task_sz);
+    } else {
+      constexpr int64_t kThreshold = 64000;
+      constexpr int64_t kChunkSize = 32000;
```

Comment:
Have you tested different thresholds?
Have you checked how it would work to split the work between available threads?

Reply:
Not yet. You mean divide the work into exactly as many parts as threads?

Reply:
Yes, that is what I'm curious about.
Maybe having this configurable would be the best option. Don't know TBH. Different things may work differently on different CPUs (AMD vs Intel).

Reply:
I think this approach (current one) might be too much. I think I will change to something like:
```
blocks_per_sample = std::min(1, num_threads * 10 / batch_size);
```
This will create up to `num_threads * 10` for batch_size=1, and won't subdivide if the batch size is big